### PR TITLE
Add more fields to NIC

### PR DESCRIPTION
**Pull request by jak3kaj**
Added the following to the `NIC` struct:
* `Speed`
* `Duplex`
* `SupportedLinkModes`
* `SupportedPorts`
* `SupportedFECModes`
* `AdvertisedLinkModes`
* `AdvertisedFECModes`

Refactoring PR #335

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -711,10 +711,25 @@ Each `ghw.NIC` struct contains the following fields:
...
 * `ghw.NIC.Capabilities` is an array of pointers to `ghw.NICCapability` structs
   that can describe the things the NIC supports. These capabilities match the
-  returned values from the `ethtool -k <DEVICE>` call on Linux
+  returned values from the `ethtool -k <DEVICE>` call on Linux as well as the 
+  AutoNegotiation and PauseFrameUse capabilities from `ethtool`.
 * `ghw.NIC.PCIAddress` is the PCI device address of the device backing the NIC.
   this is not-nil only if the backing device is indeed a PCI device; more backing
   devices (e.g. USB) will be added in future versions.
+* `ghw.NIC.Speed` is a string showing the current link speed.  This 
+  field will be present even if ethtool is not available.
```
Comment (jaypipes) [inline README.md:720]:
```suggestion
* `ghw.NIC.Speed` is a string showing the current link speed.  On Linux, this 
  field will be present even if `ethtool` is not available.
```

File: README.md [hunk 2]
```diff
@@ -711,10 +711,25 @@ Each `ghw.NIC` struct contains the following fields:
...
+* `ghw.NIC.Duplex` is a string showing the current link duplex.  This 
+  field will be present even if ethtool is not available.
```
Comment (jaypipes) [inline README.md:722]:
```suggestion
* `ghw.NIC.Duplex` is a string showing the current link duplex. On Linux, this 
  field will be present even if `ethtool` is not available.
```

**Review by jaypipes — APPROVED** [review]
Fantastic work on this @jak3kaj, really appreciate your effort! :)

File: pkg/net/net_linux.go [hunk 3]
```diff
@@ -220,3 +249,129 @@ func netDevicePCIAddress(netDevDir, netDevName string) *string {
...
+		msg := fmt.Sprintf("could not grab NIC link info for %s: %s", dev, err)
+		ctx.Warn(msg)
+		return err
+	}
+
+	m := parseNicAttrEthtool(&out)
+	nic.updateNicAttrEthtool(m)
+
+	return nil
+}
```
Comment (jaypipes) [inline pkg/net/net_linux.go:283]: @jak3kaj I think you just need to remove this function.

**Review by jaypipes — CHANGES_REQUESTED** [review]

**Review by jaypipes — COMMENTED** [review]
@jak3kaj how's it going? I wanted to know if you think you can fix up this PR today? I was looking to cut a new release of `ghw` today and wanted to get this excellent work into the release :)

I think the only thing you need to do is rebase and remove the now-uncalled functions.

File: pkg/net/net_linux.go (hunk 3 above)
Comment (jak3kaj) [reply pkg/net/net_linux.go]: Yes, I refactored this function out and forgot to remove it.

**Review by jaypipes — APPROVED** [review]
I'm good with this, thank you @jak3kaj! Could probably place that `readFile` helper method into the util pkg long-term but I won't hold this up for that request.

@ffromani if you are good with this after review, please feel free to merge.

**Comment by ffromani** [comment]
> I'm good with this, thank you @jak3kaj! Could probably place that `readFile` helper method into the util pkg long-term but I won't hold this up for that request.
> 
> @ffromani if you are good with this after review, please feel free to merge.

Looking at this ASAP

**Comment by jak3kaj** [comment]
> I'm good with this, thank you @jak3kaj! Could probably place that `readFile` helper method into the util pkg long-term but I won't hold this up for that request.
> 
> @ffromani if you are good with this after review, please feel free to merge.

I'm working on the 3rd PR to add support for the WOL data.  I can move `readFile` to utils there as well.  I got busy with something else, so this will be in a week or so.

**Review by ffromani — COMMENTED** [review]
(VERY LATE to the party but...) LGTM, thanks!

**Pull request merged**